### ISSUE TITLE
[4.2.beta3] Editor crashes when selecting Dock Position

### ISSUE THREAD

**Issue by chutneyio** [godotengine/godot#83989]
### Godot version

v4.2.beta3.mono.official [e8d57afae]

### System information

Godot v4.2.beta3.mono - Windows 10.0.22621 - Vulkan (Forward+) - integrated Intel(R) UHD Graphics (Intel Corporation; 26.20.100.7926) - Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz (8 Threads)

### Issue description

Selecting Dock Position for a scene tab after dragging it around causes the editor to crash.
Not happen on latest stable version.

### Steps to reproduce

- Reset editor layout to default (Editor -> Editor Layout -> Default)
- Drag the Scene tab to the right side
- Move it back to the left using Dock Position dropdown menu -> Crash

https://github.com/godotengine/godot/assets/145855391/8cc071f6-4b32-45ee-b96f-cf78d80519ed

### Minimal reproduction project

N/A

**Comment by chutneyio** [comment]
Update: Dragging editor tabs around seems to be broken in Godot 4.2.beta3. Here is another issue where the "Scene" and "Import" tab are not swapped despite the tab labels are changed:

https://github.com/godotengine/godot/assets/145855391/e55fa5c8-f1bd-4311-8a55-83a4b3c243eb

**Comment by akien-mga** [comment]
I can reproduce the issue in 4.2-beta3, but not in 2564026bccb74f0efa0a69b8df05bd3dea18e24d.

It seems to have been fixed already by #83966.

This was the stacktrace:
```
ERROR: Parameter "p_child" is null.
   at: remove_child (./scene/main/node.cpp:1435)
ERROR: Parameter "p_child" is null.
   at: add_child (./scene/main/node.cpp:1409)
ERROR: Parameter "p_child" is null.
   at: move_child (./scene/main/node.cpp:371)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.2.beta.custom_build (55ec171038ecc853576e646ac271450bf0ec1f8e)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x36960) [0x7faab4567960] (??:0)
[2] StringName::StringName(StringName const&) (/home/akien/Projects/godot/godot.git/./core/string/string_name.cpp:197)
[3] Node::get_name() const (/home/akien/Projects/godot/godot.git/./scene/main/node.cpp:1149)
[4] EditorNode::_save_docks_to_config(Ref<ConfigFile>, String const&) (/home/akien/Projects/godot/godot.git/./editor/editor_node.cpp:4945)
[5] EditorNode::_save_editor_layout() (/home/akien/Projects/godot/godot.git/./editor/editor_node.cpp:4933)
[6] EditorNode::_dock_select_input(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/./editor/editor_node.cpp:4799)
[7] void call_with_variant_args_helper<EditorNode, Ref<InputEvent> const&, 0ul>(EditorNode*, void (EditorNode::*)(Ref<InputEvent> const&), Variant const**, Callable::CallError&, IndexSequence<0ul>) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:303 (discriminator 4))
[8] void call_with_variant_args<EditorNode, Ref<InputEvent> const&>(EditorNode*, void (EditorNode::*)(Ref<InputEvent> const&), Variant const**, int, Callable::CallError&) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:418)
[9] CallableCustomMethodPointer<EditorNode, Ref<InputEvent> const&>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Projects/godot/godot.git/./core/object/callable_method_pointer.h:105)
[10] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Projects/godot/godot.git/./core/variant/callable.cpp:57)
[11] Object::emit_signalp(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot.git/./core/object/object.cpp:1127)
[12] Node::emit_signalp(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot.git/./scene/main/node.cpp:3607)
[13] Error Object::emit_signal<Ref<InputEvent> >(StringName const&, Ref<InputEvent>) (/home/akien/Projects/godot/godot.git/./core/object/object.h:920)
[14] Control::_call_gui_input(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/./scene/gui/control.cpp:1797)
[15] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/./scene/main/viewport.cpp:1596)
[16] Viewport::_gui_input_event(Ref<InputEvent>) (/home/akien/Projects/godot/godot.git/./scene/main/viewport.cpp:1825 (discriminator 3))
[17] Viewport::push_input(Ref<InputEvent> const&, bool) (/home/akien/Projects/godot/godot.git/./scene/main/viewport.cpp:3185)
[18] Window::_window_input(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/./scene/main/window.cpp:1551)
[19] void call_with_variant_args_helper<Window, Ref<InputEvent> const&, 0ul>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, Callable::CallError&, IndexSequence<0ul>) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:303 (discriminator 4))
[20] void call_with_variant_args<Window, Ref<InputEvent> const&>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, int, Callable::CallError&) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:418)
[21] CallableCustomMethodPointer<Window, Ref<InputEvent> const&>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Projects/godot/godot.git/./core/object/callable_method_pointer.h:105)
[22] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Projects/godot/godot.git/./core/variant/callable.cpp:57)
[23] Variant Callable::call<Ref<InputEvent> >(Ref<InputEvent>) const (/home/akien/Projects/godot/godot.git/./core/variant/variant.h:847)
[24] DisplayServerX11::_dispatch_input_event(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/platform/linuxbsd/x11/display_server_x11.cpp:4001)
[25] DisplayServerX11::_dispatch_input_events(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot.git/platform/linuxbsd/x11/display_server_x11.cpp:3978)
[26] Input::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/akien/Projects/godot/godot.git/./core/input/input.cpp:750)
[27] Input::flush_buffered_events() (/home/akien/Projects/godot/godot.git/./core/input/input.cpp:1012)
[28] DisplayServerX11::process_events() (/home/akien/Projects/godot/godot.git/platform/linuxbsd/x11/display_server_x11.cpp:5072)
[29] OS_LinuxBSD::run() (/home/akien/Projects/godot/godot.git/platform/linuxbsd/os_linuxbsd.cpp:931)
[30] godot-git(main+0x15a) [0x548b700] (/home/akien/Projects/godot/godot.git/platform/linuxbsd/godot_linuxbsd.cpp:76)
[31] /lib64/libc.so.6(+0x236b7) [0x7faab45546b7] (??:0)
[32] /lib64/libc.so.6(__libc_start_main+0x85) [0x7faab4554775] (??:0)
[33] godot-git(_start+0x21) [0x548b4e1] (??:?)
-- END OF BACKTRACE --
================================================================
```
